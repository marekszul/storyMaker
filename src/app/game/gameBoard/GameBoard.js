import React, { Component } from 'react';
import Choice from './Choice.js'

class GameBoard extends Component {
  constructor(props) {
    super(props)
    this.state =  {
      gameData : this.props.gameData,
      currentScreen : {},
      screenMap: this.createScreenMap(this.props.gameData.screens)
    }
  }
  createScreenMap(screens) {
    const screenMap = {}
    screens.forEach((s) => {
        screenMap[s.id] = s
    })
    return screenMap
  }
  setCurrentScreen (id) {
    const screen = this.state.screenMap && this.state.screenMap.hasOwnProperty(id) ? this.state.screenMap[id] : null;
    if(!screen)
      return
    this.setState({
      currentScreen: JSON.parse(JSON.stringify(screen))
    })
  }
  componentDidMount() {
    this.setCurrentScreen(this.state.gameData.startScreen)
  }
  render () {
    const imageSrc = this.state.currentScreen.hasOwnProperty("imageSrc") ? this.state.currentScreen.imageSrc : ""
    const description = this.state.currentScreen.hasOwnProperty("description") ? this.state.currentScreen.description : ""
    const options = this.state.currentScreen.hasOwnProperty("options") ? this.state.currentScreen.options : []
    const choices = [];
    for(let option of options){
        choices.push((<Choice key={'Choice_'+option.id} id={option.id} description={option.description} onClick={() => this.setCurrentScreen(option.targetScreenId)}></Choice>))
    }
    return(<div id="game_screen" class="container"> 
        <div id="game_image">
           <img alt="screen_image" src={imageSrc} width="600" height="400"></img>
           <div>
             {description}
           </div>
           <div>
             {choices}
           </div>
        </div>

    </div>)
  }
}

export default GameBoard;
