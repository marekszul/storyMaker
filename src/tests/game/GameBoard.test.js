import React from 'react'
import renderer from 'react-test-renderer'
import fs from 'fs'
import GameBoard from '../../app/game/gameBoard/GameBoard.js';

describe("GameBoard tests", () => {
    let gameState = ''
    beforeAll(() => {
        gameState = JSON.parse(fs.readFileSync('./src/tests/game/sampleGameData.json', 'utf8'))
    })
    test("Should render correctly", () => {
        console.log(gameState)
        let component = renderer.create(<GameBoard gameData={gameState}></GameBoard>)
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})