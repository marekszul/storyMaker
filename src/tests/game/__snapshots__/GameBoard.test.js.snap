// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoard tests Should render correctly 1`] = `
<div
  class="container"
  id="game_screen"
>
  <div
    id="game_image"
  >
    <img
      height="400"
      src="someImage.jpg"
      width="600"
    />
    <div>
      some story here
    </div>
    <div>
      <div>
        <button
          onClick={[Function]}
        >
          Option 
          A
        </button>
        Option1
      </div>
      <div>
        <button
          onClick={[Function]}
        >
          Option 
          B
        </button>
        Option2
      </div>
      <div>
        <button
          onClick={[Function]}
        >
          Option 
          C
        </button>
        Option3
      </div>
    </div>
  </div>
</div>
`;
